src/Core/Resolvers/PostgreSqlExecutor.cs

```diff
@@ -27,24 +28,29 @@ public class PostgreSqlQueryExecutor : QueryExecutor<NpgsqlConnection>
         /// <summary>
         /// The managed identity Access Token string obtained
         /// from the configuration controller.
+        /// Key: datasource name, Value: access token for this datasource.
         /// </summary>
-        private readonly string? _accessTokenFromController;
+        private readonly Dictionary<string, string?> _accessTokensFromConfiguration;
```

Comment:
NIT: Would it be simpler/efficient to have a single map with complex value rather than 3 different maps?

Reply:
Discussed offline. Good to have this will put in a task and move to later pr: https://github.com/Azure/data-api-builder/issues/1651